app/controllers/galleries_controller.rb

```diff
@@ -41,4 +41,16 @@ def update
     redirect_to "/galleries/#{gallery.id}"
   end
 
+  def delete
+    gallery = Gallery.find(params[:id])
+    pieces = gallery.pieces.all
+
+    pieces.each do |piece|
```

Comment:
This looks good, though you could do this by using ActiveRecord :dependent on the has_many :child association. This would allow for destroying all children when destroying the parent.